include/boost/json/storage_ptr.hpp

```diff
@@ -337,8 +335,9 @@ class storage_ptr
         storage_ptr&& other) noexcept
     {
         release();
-        i_ = detail::exchange(
-            other.i_, 0);
+        i_ = detail::exchange(other.i_,
+            reinterpret_cast<
+                std::uintptr_t>(get_default()));
```

Comment:
Should `get_default` maybe return `std::uintptr_t`? It would clean up these casts.

Reply:
Yup, did that